File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int l=0,m=0,r=nums.size()-1;
        while(m<=r) {
            if(nums[m]==0) {
                swap(nums[m],nums[l]);
                l++;
                m++;
            }
            else if(nums[m]==1) {
                m++;
            }
            else {
                swap(nums[m],nums[r]);
                r--;
            }
        }
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int n = nums.length;

        Map<Integer, Integer> map = new TreeMap<>();
        for(int num: nums) map.put(num, map.getOrDefault(num,0)+1);

        int pos = 0;
        for(Map.Entry<Integer, Integer> entry: map.entrySet()) {
            int num = entry.getKey();
            int cnt = entry.getValue();

            for(int i=0; i<cnt; i++) {
                nums[pos++] = num;
            }
        }
    }
}